### PR TITLE
feat: Merge AutoCompleteLocation inputs

### PR DESCRIPTION
**Pull request by MrSltun**
This PR resolves []

### Description
This PR is a continuation of the work to merge the different AutoComplete Location inputs throughout Eigen
The idea behind it is to merge the look and feel of 3 different inputs with the same functionality

So I decided to go with a new design for all the inputs and make sure to keep the same functionality for all fields, take your time reviewing it 🙌 

I also added a table of screenshots and videos for the old look and the new look, please take a look at it! 👀 

### Old Look
<details>
<summary>Screenshots</summary>

|  | Android | iOS |
| -- | -- | -- |
| Sell With Artsy | swa-android-old | swa-ios-old | 
| Edit My Profile | profile-android-old | profile-ios-old |
| Contact Gallery Shipping Modal | shipping-android-old | shipping-ios-old |
| Contact Gallery Shipping Modal (with Results) |  shipping-android-old-e | shipping-ios-e-old |

</details>

<details>
<summary>Videos</summary>

https://user-images.githubusercontent.com/20655703/183617155-a11d6eed-3b23-4825-8af0-ad6dff60395b.mp4

https://user-images.githubusercontent.com/20655703/183617172-e4225681-9b57-4336-aa6f-9e57158a3737.mp4

https://user-images.githubusercontent.com/20655703/183617189-fa94f374-02d3-424a-8b09-a37806dc67f9.mp4

https://user-images.githubusercontent.com/20655703/183617194-4cb7b02d-d616-4859-95b4-3a9832f4c0bc.mp4

https://user-images.githubusercontent.com/20655703/183617210-b4abf44a-7e4b-4a7e-8437-11d341afa7ec.mp4

https://user-images.githubusercontent.com/20655703/183617214-5e9dec3b-79e0-4576-b1f3-cb80e3dd42fb.mp4

</details>

## New Look

<details>
<summary>Screenshots</summary>

|  | Android | iOS |
| -- | -- | -- |
| Sell With Artsy | swa-android-new | swa-ios-new | 
| Edit My Profile | profile-android-new | profile-ios-new |
| Contact Gallery Shipping Modal | shipping-android-new | shipping-ios-new |
| Contact Gallery Shipping Modal (with Results) | shipping-android-new-e | shipping-ios-e-new |

</details>

<details>
<summary>Videos</summary>

https://user-images.githubusercontent.com/20655703/183615183-e4969aa0-f285-456a-b126-3b92d9d5b25d.mp4

https://user-images.githubusercontent.com/20655703/183615196-51996382-b143-42bb-b1eb-57a92337902a.mp4

https://user-images.githubusercontent.com/20655703/183615205-30c44c09-1008-4f29-b1be-09f2e79366e4.mp4

https://user-images.githubusercontent.com/20655703/183615208-fd658103-2696-4798-a220-79301dd539e3.mp4

https://user-images.githubusercontent.com/20655703/183615213-753843bd-4ca5-44de-9a58-d25b0b97250c.mp4

https://user-images.githubusercontent.com/20655703/183615216-448ce174-edbf-40c2-b9f0-85f53269ff40.mp4

</details>

### QA Test Case(s)

| Test Case | Feature | Environment | Acceptance Criteria | Setup Instructions/Link |
| --------- | :-----: | ----------: | ------------------: | ----------------------: |
|           |         |             |                     |                         |

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

### Follow up
- [ ] Create storybook stories for the new component
- [ ] Add the new component into Palette

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- unify autocomplete location inputs - mrsltun

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/CommercialButtons/ShippingModal.tsx [hunk 1]
```diff
@@ -28,18 +27,28 @@ export const ShippingModal: React.FC<ShippingModalProps> = (props) => {
         }}
         rightButtonText="Apply"
         onRightButtonPress={async () => {
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/CommercialButtons/ShippingModal.tsx:29]: I think we can get rid of this async here now

**Review by gkartalis — COMMENTED** [review]
Looks great with the first check! Maybe we could add some demo videos of the affected screens to make sure nothing is breaking and fix the tests and 🔥🔥🔥🔥

Amazing work!

**Review by olerichter00 — COMMENTED** [review]
Looks great!

I remember that one of the autocomplete components allowed adding custom addresses (probably `DetailedLocationAutocomplete` for the profile) and the other one only allowed adding address suggestions.

**Review by olerichter00 — APPROVED** [review]
Great work 🌟 
I've tested in on a phone and a tablet and everything works as expectedd 🌟 🌟 🌟

**Review by dariakoko — DISMISSED** [review]
Looks great! ⭐x5

File: src/app/Scenes/MyProfile/MyProfileEditForm.tsx [hunk 2]
```diff
@@ -136,7 +131,7 @@ export const MyProfileEditForm: React.FC = () => {
       initialValues: {
         name: me?.name ?? "",
         displayLocation: { display: buildLocationDisplay(me?.location || null) },
-        location: null,
+        location: me?.location || null,
```
Comment (pvinis) [inline src/app/Scenes/MyProfile/MyProfileEditForm.tsx:134]:
```suggestion
        location: me?.location ?? null,
```
same?

**Review by pvinis — APPROVED** [review]
nice one

**Review by patrinoua — APPROVED** [review]
Thanks so much for this work Sultan! 🤩 🥳
It's a great step towards consistency throughout the app!

Since this is a component we often use, and it's also added on our [Design System](https://www.figma.com/file/gZNkyqLT8AU3T61tluVJyB/Artsy-3.1-Design-System?node-id=604%3A1770) by Jimmy, I would suggest we already go ahead and add it in **palette**. 

I'd also suggest creating a story file for it so that people can see a visualisation of the different options that exist if they want to use it :)

Thanks again, you rock! 🙌🏻

File: src/app/Components/LocationAutocomplete.tests.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
+import { fireEvent } from "@testing-library/react-native"
```
Comment (MounirDhahri) [inline src/app/Components/LocationAutocomplete.tests.tsx:1]: **praise:** nice tests

File: src/app/Components/LocationAutocomplete.tsx [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+      setPredictions([])
+    } else {
+      ;(async () => {
+        const googlePredictions = await getLocationPredictions(query)
+        setPredictions(googlePredictions)
+      })()
+    }
+  }, [query])
+
+  const handleChange = async (selection?: SimpleLocation | null) => {
```
Comment (MounirDhahri) [inline src/app/Components/LocationAutocomplete.tsx:76]:
```suggestion
  const handleChange = (selection?: SimpleLocation | null) => {
```

File: src/app/Components/LocationAutocomplete.tsx [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    if (!selection?.id || !selection?.name) {
+      return
+    }
+
+    const locationDetails = await getLocationDetails(selection)
+
+    onChange(locationDetails)
+  }
```
Comment (MounirDhahri) [inline src/app/Components/LocationAutocomplete.tsx:84]: **suggestion:** maybe we can wrap this in a useEffect

**Review by MounirDhahri — APPROVED** [review]
Nice! thanks for uniting tall the location components @MrSltun . I am wondering if you QA'd the data sent as well. I remember that we spoke about how the location is sometimes an object and sometimes a string in different locations and hope you had that already in mind

**Comment by MrSltun** [comment]
> Nice! thanks for uniting tall the location components @MrSltun . I am wondering if you QA'd the data sent as well. I remember that we spoke about how the location is sometimes an object and sometimes a string in different locations and hope you had that already in mind

@MounirDhahri Yeah the first thing that I had in mind when merging the inputs was making sure that the data that is being sent is exactly the same as it used to be, and I had to make some adjustments in each screen to take in a unified data format as an output from the new input and use it to fit the requirements of the fields on the different screens

**Comment by MounirDhahri** [comment]
awesome! thanks @MrSltun
Let's get this merged then 🙌 and QA it on Friday

**Comment by patrinoua** [comment]
> Thanks so much for this work Sultan! 🤩 🥳 It's a great step towards consistency throughout the app!
> 
> Since this is a component we often use, and it's also added on our [Design System](https://www.figma.com/file/gZNkyqLT8AU3T61tluVJyB/Artsy-3.1-Design-System?node-id=604%3A1770) by Jimmy, I would suggest we already go ahead and add it in **palette**.
> 
> I'd also suggest creating a story file for it so that people can see a visualisation of the different options that exist if they want to use it :)
> 
> Thanks again, you rock! 🙌🏻

@pvinis what do you think?

**Comment by pvinis** [comment]
I think if Jimi thinks this is part of our DS, then yea, lets add it to palette-mobile. If he thinks it's not part of our DS but is part of our "this is how we do X" with X here being "autosuggest dropdown", then we move that comp the shared components folder.

In either case, I think we can do things in a separate PR, no need to block this PR here for moving files around.

Also, for storybook stories, heeeeeeelll yea! that's a great idea. Again, I'll leave this up to sultan to do this in this PR here or a separate one 👍.

**Comment by MrSltun** [comment]
I prefer that we create the storybook stories and the Palette work on separate PRs 🙌 
For the time being, I'll merge this PR later this afternoon if no one minds :)

**Comment by patrinoua** [comment]
For sure, let's merge! 🚀

We could have a follow up PR where we move the files to palette and create storybook entry :)

Thanks again Sultan!

**Pull request merged**